src/Config.php

```diff
@@ -148,7 +173,650 @@ public static function concurrencySet(int $value): void
+                break;
+        }
+
+        $vipshome = getenv("VIPSHOME");
+        if ($vipshome) {
+            $library_location = $vipshome . "/lib/";
+        } else {
+            # rely on ffi's search
+            $library_location = "";
+        }
+
+        $library = "$library_location$library_name$library_ext";
+        Utils::debugLog("init", ["libray" => $library]);
+
+        /* FIXME ... maybe display a helpful message on failure? This will
+         * probably be the main point of failure.
+         */
+        $ffi = \FFI::cdef(<<<EOS
+            int vips_init (const char *argv0);
+            const char *vips_error_buffer (void);
+            int vips_version(int flag);
+            EOS, $library);
+
+        $result = $ffi->vips_init("");
+        if ($result != 0) {
+            $msg = $ffi->vips_error_buffer();
+            throw new Vips\Exception("libvips error: $msg");
+        }
+        Utils::debugLog("init", ["vips_init" => $result]);
+
+        # get the library version number, then we can build the API
+        self::$library_major = $ffi->vips_version(0);
+        self::$library_minor = $ffi->vips_version(1);
+        self::$library_micro = $ffi->vips_version(2);
+        Utils::debugLog("init", [
+            "libvips version" => [
+                self::$library_major,
+                self::$library_minor,
+                self::$library_micro
+            ]
+        ]);
+
+        if (!self::atLeast(8, 7)) {
+            throw new Vips\Exception("your libvips is too old -- " .
+                "8.7 or later required");
+        }
+
+        if (PHP_INT_SIZE != 8) {
+            # we could maybe fix this if it's important ... it's mostly
+            # necessary since GType is the size of a pointer, and there's no
+            # easy way to discover if php is running on a 32 or 64-bit
+            # systems (as far as I can see)
+            throw new Vips\Exception("your php only supports 32-bit ints -- " .
+                "64 bit ints required");
+        }
+
+        # the whole libvips API, mostly adapted from pyvips
+        $header = <<<EOS
+// we need the glib names for these types
+typedef uint32_t guint32;
+typedef int32_t gint32;
+typedef uint64_t guint64;
+typedef int64_t gint64;
+
+// FIXME ... need to detect 32/64 bit platform, since GType is an int
+// the size of a pointer, see PHP_INT_SIZE above
+typedef guint64 GType;
+
+typedef struct _VipsImage VipsImage;
+typedef struct _VipsProgress VipsProgress;
+typedef struct _GValue GValue;
+
+int vips_init (const char *argv0);
+int vips_shutdown (void);
+
+const char *vips_error_buffer (void);
+char *vips_error_buffer_copy (void);
+void vips_error_clear (void);
+void vips_error_freeze (void);
+void vips_error_thaw (void);
+
+int vips_version(int flag);
+
+void* g_malloc (size_t size);
+void g_free (void* data);
+
+void vips_leak_set (int leak);
+
+char* vips_path_filename7 (const char* path);
+char* vips_path_mode7 (const char* path);
+
```

Comment:
When building libvips without deprecated components, it will give these errors.
```
FFI\Exception: Failed resolving C function 'vips_path_filename7'
FFI\Exception: Failed resolving C function 'vips_path_mode7'
```
These decls are currently unused, so it is probably safe to remove them.
```suggestion

```